File: src/mk3/esp32package/TAMAF-ESP32/src/tamaf/behaviors/OneShotBehavior.cpp
#include "OneShotBehavior.h"

namespace tamaf {
namespace behaviors {

OneShotBehavior::OneShotBehavior() : Behavior() {}

OneShotBehavior::OneShotBehavior(tamaf::core::Agent* agent) : Behavior() {
    setAgent(agent);
}

bool OneShotBehavior::Done() {
    return true; // Always finishes after one execution
}

} // namespace behaviors
} // namespace tamaf

File: src/mk2/JavaPackage/src/main/java/tamaf/OneShotBehavior.java
package tamaf;

public abstract class OneShotBehavior extends Behavior {
    
    public OneShotBehavior(Agent agent) {
        super(agent);
    }

    public OneShotBehavior() {
        super();
    }
    
    @Override
    public boolean done() { return true; }
}
